packages/general/src/storage/StorageBackendMemory.ts

```diff
@@ -57,12 +57,57 @@ export class StorageBackendMemory extends SyncStorage implements CloneableStorag
         return this.store;
     }
 
-    get<T extends SupportedStorageTypes>(contexts: string[], key: string): T | undefined {
+    get(contexts: string[], key: string): SupportedStorageTypes | undefined {
         this.#assertInitialized();
         if (!contexts.length || !key.length) throw new StorageError("Context and key must not be empty.");
         return this.store[this.createContextKey(contexts)]?.[key];
     }
 
+    readBlob(contexts: string[], key: string): ReadableStream<Uint8Array> {
+        const value = this.get(contexts, key);
+        if (value === undefined) {
+            return new ReadableStream<Uint8Array>({
+                start(controller) {
+                    controller.close();
+                },
+            });
+        }
+        if (!ArrayBuffer.isView(value)) {
+            throw new StorageError("Value must be an ArrayBuffer or a typed array to read as stream.");
```

Comment:
ArrayBuffer is not a view, is this just "value is not a blob"?

Reply:
I just used the same check as we use in other places too when we want to detect if it is an Uint8array style object ... in fact thats the purpose of the check. Any better idea?

Reply:
The error is confusing because the caller of this method may not know about the underlying implementation. They asked for a blob but got an error about something unrelated

The other thing that's weird is the only thing it can legally be is a Uint8Array...  So even if the error is desirable, it should either specify Uint8Array or the code should be adjusted to support other buffer types